fbpic/openpmd_diag/field_diag.py

```diff
@@ -163,21 +165,69 @@ def write_dataset( self, field_grp, path, quantity ) :
             dset = None
 
         # Write the mode 0 : only the real part is non-zero
-        mode0 = self.get_dataset( quantity, 0 )
+        mode0 = self.get_dataset( quantity, 0)
         if self.rank == 0:
             mode0 = mode0.T
             dset[0,:,:] = mode0[:,:].real
         # Write the higher modes
         # There is a factor 2 here so as to comply with the convention in
         # Lifschitz et al., which is also the convention adopted in Warp Circ
         for m in range(1,self.fld.Nm):
-            mode = self.get_dataset( quantity, m )
+            mode = self.get_dataset( quantity, m)
             if self.rank == 0:
                 mode = mode.T
                 dset[2*m-1,:,:] = 2*mode[:,:].real
                 dset[2*m,:,:] = 2*mode[:,:].imag
 
-    def get_dataset( self, quantity, m ):
+    def write_dataset_envelope( self, field_grp, path, quantity):
+        """
+        Write a given dataset
+
+        Parameters
+        ----------
+        field_grp : an h5py.Group object
+            The group that corresponds to the path indicated in meshesPath
+
+        path : string
+            The relative path where to write the dataset, in field_grp
+
+        quantity : string
+            Describes which envelope is being written.
+            (Either A or dtA)
+        """
+        path_real = "%s_real" %(path)
+        path_imag = "%s_imag" %(path)
```

Comment:
I know that this function is always called with `path` equal to `quantity`. 
However, could you maybe do `%(quantity)` instead as `%(path)`, as it is maybe more intuitive?